provider/disk_utilities.py

```diff
@@ -0,0 +1,57 @@
+"""
+Disk utilities functions for managing disks and filesystems in guest environments.
+
+This module contains functions related to disk detection, filesystem creation,
+and command execution for various OS types.
+"""
+
+from virttest import error_context, utils_disk, utils_misc
+from virttest.utils_misc import get_linux_drive_path
+from virttest.utils_windows.drive import get_disk_props_by_serial_number
+
+
+def get_window_disk_index_by_serial(serial, session):
+    idx_info = get_disk_props_by_serial_number(session, serial, ["Index"])
+    if idx_info:
+        return idx_info["Index"]
+
+
+def get_disk_and_prepare_cmd(guest_cmd, img, params, session, test):
+    os_type = params["os_type"]
+    error_context.context(f"Check disk {img} in guest", test.log.debug)
+
+    img_size = params.get(f"image_size_{img}")
+
+    if os_type == "windows":
+        disk = get_window_disk_index_by_serial(img, session)
+        if disk:
+            utils_disk.update_windows_disk_attributes(session, disk)
+            error_context.context(f"Cleaning disk: {disk}", test.log.info)
+            utils_disk.clean_partition_windows(session, disk)
+            error_context.context(f"Formatting disk: {disk}", test.log.info)
+            driver = utils_disk.configure_empty_disk(session, disk, img_size, os_type)[
+                0
+            ]
+            output_path = f"{driver}:\\test.dat"
+            guest_cmd = utils_misc.set_winutils_letter(session, guest_cmd).format(
+                output_path
+            )
+    else:
+        output_path = get_linux_drive_path(session, img)
+        guest_cmd = guest_cmd.format(output_path)
+
+    return guest_cmd
+
+
+def guest_fs_mount_and_dd_cmd(os_type, tmp_dir):
```

Comment:
same thinking. we may consider more powerful encapsulation, like as 

execute_cmd_on_disk_by_id():
 drv= init_disk_by_id
 cmd=xxxxx
session_cmd ()

the utils provide different granularity functions, most cases may use execute_cmd_on_disk_by_id directly.